test/libsolidity/util/BytesUtils.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/util/BytesUtils.h>
+
+#include <liblangutil/Common.h>
+
+#include <boost/algorithm/string.hpp>
+
+#include <fstream>
+#include <memory>
+#include <regex>
+#include <stdexcept>
+
+using namespace dev;
+using namespace langutil;
+using namespace solidity;
+using namespace dev::solidity::test;
+using namespace std;
+using namespace soltest;
+
+bytes BytesUtils::convertBoolean(string const& _literal)
+{
+	if (_literal == "true")
+		return bytes{true};
+	else if (_literal == "false")
+		return bytes{false};
+	else
+		throw Error(Error::Type::ParserError, "Boolean literal invalid.");
+}
+
+bytes BytesUtils::convertNumber(string const& _literal)
+{
+	try
+	{
+		return toCompactBigEndian(u256{_literal});
+	}
+	catch (std::exception const&)
+	{
+		throw Error(Error::Type::ParserError, "Number encoding invalid.");
+	}
+}
+
+bytes BytesUtils::convertHexNumber(string const& _literal)
+{
+	try
+	{
+		if (_literal.size() % 2)
+			throw Error(Error::Type::ParserError, "Hex number encoding invalid.");
+		else
+			return fromHex(_literal);
+	}
+	catch (std::exception const&)
+	{
+		throw Error(Error::Type::ParserError, "Hex number encoding invalid.");
+	}
+}
+
+bytes BytesUtils::convertString(string const& _literal)
+{
+	try
+	{
+		return asBytes(_literal);
+	}
+	catch (std::exception const&)
+	{
+		throw Error(Error::Type::ParserError, "String encoding invalid.");
+	}
+}
+
+string BytesUtils::formatUnsigned(bytes const& _bytes) const
+{
+	stringstream os;
+
+	if (*_bytes.begin() & 0x80)
+		os << u2s(fromBigEndian<u256>(_bytes));
+	else
+		os << fromBigEndian<u256>(_bytes);
+
+	return os.str();
+}
+
+string BytesUtils::formatSigned(bytes const& _bytes) const
+{
+	stringstream os;
+
+	if (*_bytes.begin() & 0x80)
```

Comment:
What if `_bytes` is empty?

Reply:
Added an assertion since this function is meant to be called with an non-empty bytes only.